src/lm.jl

```diff
@@ -114,23 +114,30 @@ LinearAlgebra.cholesky(x::LinearModel) = cholesky(x.pp)
 function StatsBase.fit!(obj::LinearModel)
     if isempty(obj.rr.wts)
         delbeta!(obj.pp, obj.rr.y)
-    else 
+    else
         delbeta!(obj.pp, obj.rr.y, obj.rr.wts)
     end
-    installbeta!(obj.pp)     
+    installbeta!(obj.pp)
     updateμ!(obj.rr, linpred(obj.pp, zero(eltype(obj.rr.y))))
     return obj
 end
 
 function fit(::Type{LinearModel}, X::AbstractMatrix{<:Real}, y::AbstractVector{<:Real},
-             allowrankdeficient::Bool=false; wts::AbstractVector{<:Real}=similar(y, 0))
+             allowrankdeficient_dep::Union{Bool,Nothing}=nothing;
+             wts::AbstractVector{<:Real}=similar(y, 0),
+             allowrankdeficient::Bool=false)
+    if allowrankdeficient_dep != nothing
+        @warn "Positional argument `allowrankdeficient` is deprecated, use keyword" *
+              "argument instead. Proceeding with positional argument  value: $allowrankdeficient_dep"
+              allowrankdeficient = allowrankdeficient_dep
```

Comment:
```suggestion
    if allowrankdeficient_dep != nothing
        Base.depwarn("Positional argument `allowrankdeficient` is deprecated, use keyword" *
                     "argument instead. Proceeding with positional argument  value: $allowrankdeficient_dep",
                     :fit)
              allowrankdeficient = allowrankdeficient_dep
```

Reply:
I couldn't get this to work, I'll try again in a bit.

Reply:
Got it to work!